C7/Game.cs

```diff
@@ -440,6 +442,15 @@ enum GameState {
 
 	// Handle Godot keybind actions
 	private void processActions() {
+		if (Input.IsActionJustPressed(C7Action.Escape) && popupOverlay.Visible) {
+			popupOverlay.OnHidePopup();
+			return;
+		}
+
+		// never poll for actions if UI elements are visible
+		if (popupOverlay.Visible) {
```

Comment:
what's the distinction between Visible and ShowingPopup()?

Reply:
hmm I wrote ShowingPopup to use here but I must have forgotten to change it... currently there is no difference aside from PopupOverlay.ShowingPopup signalling clear intent vs. the overlay itself being visible which is less clear imo. Will fix

Reply:
bump